app/javascript/src/NewService/components/NewServiceForm.jsx

```diff
@@ -0,0 +1,150 @@
+// @flow
+
+import React, {useState, useEffect} from 'react'
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+import 'core-js/es6/promise'
+import {ServiceSourceForm, FormTemplate} from 'NewService'
+import type {SourceFormProps, FormTemplateProps} from 'NewService/types'
+
+import {createReactWrapper} from 'utilities/createReactWrapper'
+
+const BASE_PATH = '/p/admin/service_discovery'
+const PROJECTS_PATH = `${BASE_PATH}/projects.json`
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+type Response = {
+  ok: boolean,
+  statusText: string,
+  json: () => Promise<void>
+}
+const fetchData = async (url: string, setError) => {
+  setError({hasError: false, errorMsg: ''})
+  const namespace = url === PROJECTS_PATH
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url)
+    .then((response: Response) => {
+      if (!response.ok) {
+        setError({hasError: true, errorMsg: response.statusText})
+      }
+      return response.json()
+    })
+    .then(data => data[namespace].reduce((acc, current) => acc.concat(current.metadata.name), []))
+    .catch(error => {
+      setError({hasError: true, errorMsg: error.message})
+    })
+}
+
+const callEffect = (servicesRequested, selectedNamespace, startServicesFetch) => {
+  if (servicesRequested && selectedNamespace) {
+    startServicesFetch(`${BASE_PATH}/namespaces/${selectedNamespace}/services.json`)
+  }
+}
+
+const renderForm = (sourceFormProps, serviceDiscoveryProps, serviceNewProps) =>
+  <React.Fragment>
+    <h1>New API</h1>
+    <ServiceSourceForm {...sourceFormProps}/>
```

Comment:
I'd prefer not, because enzyme still doesn't have full support for hooks, including `useContext`. 
It's true it could help with the props problem, but testing will be more complicated.
As enzyme support for hooks is an [open discussion](https://github.com/airbnb/enzyme/pull/2041), it may change a lot. So I think it's better to wait until we have something more stable.
In the meantime I'd stay with this approach, even if it's more verbose